crates/static-analysis-kernel/src/analysis/javascript.rs

```diff
@@ -39,6 +113,7 @@ pub fn execute_rule(
     filename: String,
     analysis_options: AnalysisOptions,
     file_context: &FileContext,
+    tree: ResourceTree,
```

Comment:
Instead of passing the tree, we should have a structure like `JavascriptRuntimeContext` or `JavascriptExecutionContext` that will contain all the data/values we need. This way, if we need more data/values in the JavaScript context, we just need to add more attribute in this type/structure.